### ISSUE TITLE
Unit option parser handles newlines differently than before

### ISSUE THREAD

**Issue by arianvp** [systemd/systemd#12898]
**systemd version the issue has been seen with**
> v242

**Used distribution**
 > NixOS-unstable
> Upstream issue: https://github.com/NixOS/nixpkgs/issues/63533

**Expected behaviour you didn't see**
```
ExecStart=a \
          b \
          c \
           \
          d \
           \
          

ExecStartPre=e
```
parses as:

```
ExecStart="a b c d"
ExecStartPre="e"
```

**Unexpected behaviour you saw**
It parses as:
```
ExecStart="a b c d ExecStartPre=e"
```
**Steps to reproduce the problem**
See the above unit definition

This difference in parsing occured somewhere between `v239` and `v242` as we just migrated NixOS to `v242` from `v239`.  We generate a lot of systemd units with scripts, and we have identified a few units that stopped working due to this small change in parsing behaviour.
I was wondering if this stricter parsing rule is a bug or a feature, and if it is a feature, which commit introduced it? I have not found clear candidates whilst glancing over the git history

**Comment by yuwata** [comment]
Duplicat of #12883 and already fixed by #12890.

**Comment by arianvp** [comment]
Thanks for the quick response!
